frame/support/src/storage/mod.rs

```diff
@@ -811,16 +815,292 @@ mod private {
 
 	impl<T: Encode> Sealed for Vec<T> {}
 	impl<Hash: Encode> Sealed for Digest<Hash> {}
+	impl<T: Value, S: Get<u32>> Sealed for BoundedVec<T, S> {}
 }
 
 impl<T: Encode> StorageAppend<T> for Vec<T> {}
 impl<T: Encode> StorageDecodeLength for Vec<T> {}
 
-/// We abuse the fact that SCALE does not put any marker into the encoding, i.e.
-/// we only encode the internal vec and we can append to this vec. We have a test that ensures
-/// that if the `Digest` format ever changes, we need to remove this here.
+/// We abuse the fact that SCALE does not put any marker into the encoding, i.e. we only encode the
+/// internal vec and we can append to this vec. We have a test that ensures that if the `Digest`
+/// format ever changes, we need to remove this here.
 impl<Hash: Encode> StorageAppend<DigestItem<Hash>> for Digest<Hash> {}
 
+/// Marker trait for types `T` that can be stored in storage as `Vec<T>`.
+pub trait Value: FullCodec + Clone + sp_std::fmt::Debug + Eq + PartialEq {}
```

Comment:
Note that this is more restrictive than necessary, we can store put in a storage value types which doesn't implement `Eq`, `PartialEq`, `Clone` or `Debug`. But I guess we are fine only supporting this.

Reply:
note that if we want to support BoundedVec for value which doesn't implement such traits, then we can implement Default, Clone, Debug manually with a where condition.

Reply:
I removed the bound for `Eq + PartialEq` from `trait Value`, it was not necessary actually. 

I also tried to remove the whole trait, but that would be rather problematic: For `BoundedVec<T>` to be `Clone` and `Debug`, we need `T: Clone` and `Debug`, and I don't like repeating this too much. I renamed the wrapper trait to: 

```
/// Marker trait for types `T` that can be stored in storage as `Vec<T>`.
pub trait BoundedVecValue: FullCodec + Clone + sp_std::fmt::Debug {}
impl<T: FullCodec + Clone + sp_std::fmt::Debug> BoundedVecValue for T {}
```

for better clarrity